File: mobile-android/app/src/main/java/com/erenlermarket/app/domain/model/Address.kt
package com.erenlermarket.app.domain.model

data class Address(
    val id: String,
    val label: String,
    val fullAddress: String,
    val city: String,
    val district: String,
    val buildingName: String = "",
    val buildingNo: String = "",
    val floor: String = "",
    val apartmentNo: String = "",
    val isDefault: Boolean,
    val latitude: Double? = null,
    val longitude: Double? = null,
) {
    val summary: String get() = "$district, $city"

    /** "Erenler Apt, No 12, Kat 3, Daire 7" — boş alanlar atlanır. */
    val buildingLine: String
        get() = listOfNotNull(
            buildingName.takeIf { it.isNotBlank() },
            buildingNo.takeIf { it.isNotBlank() }?.let { "No $it" },
            floor.takeIf { it.isNotBlank() }?.let { "Kat $it" },
            apartmentNo.takeIf { it.isNotBlank() }?.let { "Daire $it" },
        ).joinToString(", ")
}

/** Yeni adres girdisi. */
data class NewAddress(
    val label: String,
    val fullAddress: String,
    val city: String,
    val district: String,
    val buildingName: String = "",
    val buildingNo: String = "",
    val floor: String = "",
    val apartmentNo: String = "",
    val isDefault: Boolean = false,
    val latitude: Double? = null,
    val longitude: Double? = null,
)

File: mobile-ios/Modules/Domain/Sources/Entities/Address.swift
import Foundation

public struct Address: Identifiable, Equatable, Sendable {
    public let id: String
    public let label: String
    public let fullAddress: String
    public let city: String
    public let district: String
    public let isDefault: Bool
    public let latitude: Double?
    public let longitude: Double?

    public init(
        id: String,
        label: String,
        fullAddress: String,
        city: String,
        district: String,
        isDefault: Bool,
        latitude: Double? = nil,
        longitude: Double? = nil
    ) {
        self.id = id
        self.label = label
        self.fullAddress = fullAddress
        self.city = city
        self.district = district
        self.isDefault = isDefault
        self.latitude = latitude
        self.longitude = longitude
    }

    public var summary: String {
        "\(district), \(city)"
    }
}

/// Yeni adres girişi.
public struct NewAddress: Sendable, Equatable {
    public var label: String
    public var fullAddress: String
    public var city: String
    public var district: String
    public var isDefault: Bool
    public var latitude: Double?
    public var longitude: Double?

    public init(
        label: String,
        fullAddress: String,
        city: String,
        district: String,
        isDefault: Bool = false,
        latitude: Double? = nil,
        longitude: Double? = nil
    ) {
        self.label = label
        self.fullAddress = fullAddress
        self.city = city
        self.district = district
        self.isDefault = isDefault
        self.latitude = latitude
        self.longitude = longitude
    }
}
